peer.go

```diff
@@ -837,12 +846,23 @@ out:
 			// this channel, then this means that we're the
 			// responder to the workflow.
 			if _, ok := chanShutdowns[req.ChannelID]; !ok {
+				// Check responderShutdownSigs for an already
+				// existing shutdown signature for this channel.
+				// If such a signature exists, it means we
+				// already have sent a response to a shutdown
+				// message for this channel, so ignore this one.
+				_, exists := responderShutdownSigs[req.ChannelID]
```

Comment:
This is what replaces the `if channel.ChannelStatus() == lnwallet.ChannelClosing` check I removed from `CreateCloseProposal` (and initially moved to `handleShutdownResponse`). Similar to that check, this will reject any attempt to start the channel shutdown procedure twice.

I don't really know if it is needed, I just tried to keep the original behavior. Should I remove it?